### PR TITLE
minor travis improvements

### PR DESCRIPTION
**Pull request by srenatus**
* don't stick to deprecated env
* remove rvm matrix thing
* build for chef 12 and 13
* README: add badges (travis and rubygems version)

I'm not sure if the multiple version tests really help, but it doesn't seem like the worst idea. (Anyhow, it doesn't come at a high cost, I think.)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
-# Lets actually use our tests! :smile:
```
Comment (afiune) [inline .travis.yml]: Oh man, I was so happy to write this line! 😂

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+sudo: false
 branches:
   only:
     - master
 
-sudo: required
 language: ruby
 cache: bundler
 dist: trusty
+rvm: 2.3.3
```
Comment (afiune) [inline .travis.yml]: Not sure if matters but I see chefdk built with ruby version:
```
➜  ~ /opt/chefdk/embedded/bin/gem -v
2.6.11
```

File: README.md [hunk 3]
```diff
@@ -1,4 +1,6 @@
 # knife EC backup
+[![Build Status Master](https://travis-ci.org/chef/knife-ec-backup.svg?branch=master)](https://travis-ci.org/chef/knife-ec-backup)
```
Comment (afiune) [inline README.md]: Nice!!!!! 👍

File: .travis.yml (hunk 2 above)
Comment (thommay) [inline .travis.yml]: ChefDK 1.4 uses ruby 2.3; ChefDK 2 will use ruby 2.4.1. What you're showing there is the version of rubygems, not ruby :)

Comment (afiune) [inline .travis.yml]: Its 8 am and I haven't have my coffee. Thanks @thommay - 😅  #disregard

**Comment by stevendanna** [comment]
> Do we run knife-ec-backup from a workstation? (that would involve using chefdk insted?)
> Or do we run it from within the Chef Server?

Typically one runs it from the chef-server directly; however, we want to keep it installable on the workstation since it is occasionally useful to use it from there.  Also testing on the N+1 version of chef from what ships in Chef Server will help us catch problems earlier.

**Pull request merged**